### PR TITLE
Add makefile for Robotis OpenCM

### PR DESCRIPTION
**Pull request by fuzzyTew**
Here is a makefile to build sketches from the Robotis OpenCM IDE from the command-line.

Robotis CM are a small set of higher end boards useful for controlling the high-quality-but-cheap Robotis XL-320 servos, along with other devices.

### CODE REVIEW

**Comment by sej7278** [comment]
wow this is excellent!

i wonder if we should up the version number for such a big change though, @sudar 1.6 maybe?

we need to update the travis-ci bits to ignore these examples i guess see runtests.sh (or install the ide like chipkit.sh) which is why the build check fails.

would also need to update the debian/fedora packaging files to cope with the new file in bin/ which may need a manpage to pass lintian

**Comment by sudar** [comment]
@fuzzyTew 

As @sej7278 mentioned, this is really excellent. Very excited.

@sej7278 Yeah, let's increase the version to 1.6. I am assuming we will not have any breaking changes. If there will be breaking changes then we may have to increase the version to 2.0.

**Comment by fuzzyTew** [comment]
I did what the Teensy implementation did and removed my example from the tests.
I added Fedora packaging, but I'm not running Fedora to test it.  It looks like the Debian packaging is hosted elsewhere.

File: OpenCM.mk [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+ifndef ARDMK_DIR
+    ARDMK_DIR := $(realpath $(dir $(realpath $(lastword $(MAKEFILE_LIST)))))
+endif
+
+# include Common.mk now we know where it is
+include $(ARDMK_DIR)/Common.mk
+
+ARDMK_VENDOR         = robotis
+ARDUINO_CORE_PATH    = $(ARDUINO_DIR)/hardware/robotis/cores/robotis
```
Comment (sej7278) [inline OpenCM.mk]: shouldn't these be override-able?

also where is ARDUINO_DIR defined, surely the core isn't installed within the system's arduino ide?

**Comment by sej7278** [comment]
i wasn't really expecting all of my issues to be fixed straight away lol!

just tried packaging it for debian and lintian is moaning about the binary being called robotis-loader.py so can we rename the script to robotis-loader and the manpage to robotis-loader.1 which will need changes to OpenCM.mk and arduino-mk.spec as far as i can tell.

changed to upstream debian files i made are:

arduino-mk.manpages now reads:
```ard-reset-arduino.1 robotis-loader.1```

control now uses:
```Standards-Version: 4.0.0```

File: OpenCM.mk (hunk 1 above)
Comment (fuzzyTew) [inline OpenCM.mk]: I followed Teensy's model; ARDUINO_DIR was to be set to the path to the OpenCM_IDE, and all OpenCM boards are robotis.  But I'll change it!

**Comment by sej7278** [comment]
couple more tweaks, although i think @sudar runs a script to update the versions....?

tested it builds on fedora 26 now too.

```diff
diff --git a/HISTORY.md b/HISTORY.md
index 7bd7583..d8f89e6 100644
--- a/HISTORY.md
+++ b/HISTORY.md
@@ -5,13 +5,16 @@ The following is the rough list of changes that went into different versions.
 I tried to give credit whenever possible. If I have missed anyone, kindly add it to the list.
 
 ### In Development
+- TBC
+
+### 1.6.0 (2017-07-11)
 - Fix: Allowed for SparkFun's weird usb pid/vid submenu shenanigans (issue #499). (https://github.com/sej7278)
 - Fix: Do not include the Arduino header when calling generate_assembly on .cpp files. (https://github.com/Batchyx)
 - Fix: Auto-detect F_CPU on Teensy from boards.txt (https://github.com/DaWelter)
 - Fix: params typo in PuTTY section (issue #487) (https://github.com/ericdand)
 - Fix: Fixed sed expression to properly format show_submenu (issue #488) (https://github.com/cbosdo)
-- New: Add support for good old cu as monitor command (issue #492) (https://github.com/mwm)
 - Tweak: Removed tilde from documentation (issue #497). (https://github.com/sej7278)
+- New: Add support for good old cu as monitor command (issue #492) (https://github.com/mwm)
 - New: Add a documentation how to setup Makefile for 3rd party boards (issue #499). (https://github.com/MilanV)
 - New: Add support for Robotis OpenCM boards
 
diff --git a/README.md b/README.md
index e55e20e..7ea5c0b 100644
--- a/README.md
+++ b/README.md
@@ -289,7 +289,7 @@ For large Robotis projects, [libmaple](https://github.com/Rhoban/Maple) may be m
 
 ## Versioning
 
-The current version of the makefile is `1.5.2`. You can find the full history in the [HISTORY.md](HISTORY.md) file
+The current version of the makefile is `1.6.0`. You can find the full history in the [HISTORY.md](HISTORY.md) file
 
 This project adheres to Semantic [Versioning 2.0](http://semver.org/).
 
diff --git a/packaging/fedora/arduino-mk.spec b/packaging/fedora/arduino-mk.spec
index ed598ef..61bf317 100644
--- a/packaging/fedora/arduino-mk.spec
+++ b/packaging/fedora/arduino-mk.spec
@@ -1,5 +1,5 @@
 Name:                  arduino-mk
-Version:               1.5.2
+Version:               1.6.0
 Release:               1%{dist}
 Summary:               Program your Arduino from the command line
 Packager:              Simon John <git@the-jedi.co.uk>
```

**Comment by sudar** [comment]
@sej7278 

I just ran the script and the places you identified are the only places where we have to update the version.

**Comment by sej7278** [comment]
probably need a more functional example Makefile too, i just tried and its missing MONITOR_PORT and OPENCMIDE_DIR isn't defined anywhere so it won't compile or upload! this works:

```make
BOARD_TAG = cm904
OPENCMIDE_DIR = $(HOME)/ROBOTIS_OpenCM-v1.0.2
MONITOR_PORT = /dev/ttyUSB0

include ../../OpenCM.mk
```

**Comment by fuzzyTew** [comment]
The blink example uploads fine to the board for me.  `MONITOR_PORT` is autodetected when the board is actually connected.

**Comment by sej7278** [comment]
looks good to me, not sure if @sudar will want a single squashed commit?

**Comment by fuzzyTew** [comment]
I squash it but ...  I tried compiling a more complex example and ran into linker errors.  Needs a little more work.

**Comment by fuzzyTew** [comment]
To make links succeed I changed the linker in Arduino.mk from `$(CC)` to `$(CXX)`, which is what the OpenCM IDE uses.  Otherwise it failed to include symbols for some C++ destructor.  Will this change break anything else?  I could possibly find how to include the proper symbols manually, or abstract the change away so it can be selected with a further variable.

**Comment by sej7278** [comment]
yeah you can't change Arduino.mk like that.

will have to think of another way to do it. seems a bit odd it links ok in c++ mode but not c.....?

**Comment by fuzzyTew** [comment]
It was the obvious: needed libstdc++ to be linked.

Fixed and rebased to a single commit.

**Comment by sej7278** [comment]
Looks good to me 👍

**Comment by sudar** [comment]
Thanks @sej7278 @fuzzyTew for iterating on the PR and for testing it.

I will try it out over the weekend and merge it.

**Comment by sudar** [comment]
Just merged this PR to master and made a new release.

**Comment by sej7278** [comment]
could be the weird new github gui, but it looks like you've tagged it but not made a new release?

**Comment by sudar** [comment]
@sej7278 Actually I only tagged yesterday.

I just created the release in Github.

**Pull request merged**